From 6ec9368504eeaac63436f642d19a3f100e2e7504 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 14:38:23 -0400
Subject: [ethernet] enic: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate

Message-id: <1409668716-4814-12-git-send-email-sassmann@redhat.com>
Patchwork-id: 90636
O-Subject: [RHEL7.1 PATCH 11/24] net: use SPEED_UNKNOWN and DUPLEX_UNKNOWN when appropriate
Bugzilla: 1107934
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Jiri Pirko <jiri@resnulli.us>

Signed-off-by: Jiri Pirko <jiri@resnulli.us>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 537fae0101c8853abb52136788173dde74b9d1e9)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/cisco/enic/enic_ethtool.c b/drivers/net/ethernet/cisco/enic/enic_ethtool.c
index 1882db2..2e50b54 100644
--- a/drivers/net/ethernet/cisco/enic/enic_ethtool.c
+++ b/drivers/net/ethernet/cisco/enic/enic_ethtool.c
@@ -104,8 +104,8 @@ static int enic_get_settings(struct net_device *netdev,
   ethtool_cmd_speed_set(ecmd, vnic_dev_port_speed(enic->vdev));
   ecmd->duplex = DUPLEX_FULL;
  } else {
-  ethtool_cmd_speed_set(ecmd, -1);
-  ecmd->duplex = -1;
+  ethtool_cmd_speed_set(ecmd, SPEED_UNKNOWN);
+  ecmd->duplex = DUPLEX_UNKNOWN;
  }
 
  ecmd->autoneg = AUTONEG_DISABLE;
-- 
1.7.1